### DISCUSSION TITLE
MQTT to Home Assistant / Spa Temp

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#327] [Q&A]
Good afternoon – Home Assistant working great with AqualinkD. The only issue I have is that if the pool pump is running and then I turn on the spa pump it shows the proper pool temperature, but shows zero as the spa temperature which has an independent thermocouple. Is there some setting that I need to adjust for the spa temperature to read independently of the pool temperature? They are basically 2 Independent closed-Loop systems.

**Comment by sfeakes** [comment]
First check what the web UI show for the temperatures.
But the control panel will only display one water body temperature at any given time.  So if the panel is in pool mode, then pool water temp should be accurate (and spa showing 0), and if it's in spa mode, then spa water temp will be accurate and pool will be last known value.

If your control panel shows both the accurate spa and pool temp at the same time, (maybe a panel version I have not come across) then AqualinkD has not been coded for that.  If that's the case, I'll need to get some debugging information from you to see if I can make some changes.

**Reply by jmp2000** [reply]
Ah - interesting! Well, the good news is that in fact, my panel does show both the pool and spa temperature, and air temperature for that matter - independently so basically three thermisters. Let me know what information I could pass along that might make it to a future version! Thanks so much.

**Comment by sfeakes** [comment]
Go to aqualinkd's management page.  `http://aqualinkd.ip.address/aqmanager.html`

-  Let me know the Panel version and Type from under `system` on the left side.
(Turn on setting/pumps where you would see both pool and spa temperatures.)
- Expand the `debug mask` section and enable `all button`, `One Touch`, `iAQ Touch` buttons.
- Let that run for a minute or two ( hopefully you see both Pool and Spa temp messages )
- Then expand `log file` section, download the file and post it. (you might need to increase lines section depending on how much you caught/want.

**Comment by jmp2000** [comment]
Ok - followed the directions you sent and have the info below.

One thing I noticed that was interesting - in the AQManager (Web UI) - i
have a tile for:
Spa Water Temperature
Pool Water Temperature
Pool Air Temperature

With the Spa pump and Pool pump running - it only shows Spa Water
Temperature and Pool Air Temperature tiles reading data.
If I turn off the Spa Pump and only run the Pool pump - it shows Pool Water
Temperature and Pool Air Temperature reading data.
If I turn off the Pool Pump and only run the Spa pump - it shows only Pool
Air Temperature reading data
If I turn off both pumps (Spa and Pool) - it only shows Pool Air
Temperature reading data

AqualinkD version: 2.3.7
Panel Version:  E0260807 REV QQ
Panel Type: RS-6 Combo Pool/Spa Dual Equipment

Let me know if that helps - let me know what else you need!

**Reply by sfeakes** [reply]
Did you forget to upload the log?

**Comment by jmp2000** [comment]


**Reply by sfeakes** [reply]
No attachment. are you selecting the paperclip and attaching the downloaded file?

**Comment by jmp2000** [comment]
Ok here we go - apologies!
[aqualinkd.log](https://github.com/user-attachments/files/16571140/aqualinkd.log)

**Reply by sfeakes** [reply]
Thanks for the log.   Looking over the log and the code, it looks like the setting below would work for your setup.
`report_zero_spa_temp = no`
My guess is in your aqualinkd.conf, you have that set to `yes`.  
The settings was designed for a slightly different purpose, but should also work for what you are after.

**Reply by jmp2000** [reply]
Ok - made progress!  You were correct - the setting I had was to report that temp "yes".  I switched it over to "no".
Now:
Pool & Spa pumps off = Only Air temp displayed (good)
Pool pump on = Pool & Air temp displayed (good)
Spa pump on = Spa & Air temp displayed (good)
Pool pump and Spa pump both on = Spa Temp displayed & Air Temp displayed.  But no pool temp, even though the pool pump is on.  The Pool temp blinked once on Aqmanager with a reading and then stayed 'dark' with no reading since then.  Weird.   

So it's giving me good data - but just doesn't seem to be able to display all 3 temperature indicators (Air, Pool, Spa) - when both the pool and spa pump are operating.

**Reply by sfeakes** [reply]
Did you check HA?   I believe both should be showing correctly in HA & MQTT.   I think I will still need to make some changes for this to work in the web UI.

**Reply by jmp2000** [reply]
Yes, I checked in HA - still glitchy on being able to display the Air/Pool/Spa temp at the same time (and updating data) through HA when both pumps are on.

**Reply by sfeakes** [reply]
@jmp2000 Version 2.3.8 now supports multiple temp sensors

**Comment by sfeakes** [comment]
I've just posted a development update that should fix this issue 2.3.8 (Dev).
